### PR TITLE
UTF8-Patch V3 2015.09.22.11.45.00

### PR DESCRIPTION
**Pull request by BestianRU**
oleg@oleg-devel-linux-2# go build
		oleg@oleg-devel-linux-2# go test
		TestDial: starting...
		TestDial: finished...
		TestDialTLS: starting...
		TestDialTLS: finished...
		TestStartTLS: starting...
		TestStartTLS: finished...
		TestSearch: starting...
		TestSearch: (cn=cis-fac) -> num of entries = 0
		TestSearchStartTLS: starting...
		TestSearchStartTLS: (cn=cis-fac) -> num of entries = 0
		TestSearchStartTLS: upgrading with startTLS
		TestSearchStartTLS: (cn=cis-fac) -> num of entries = 0
		TestSearchWithPaging: starting...
		TestSearchWithPaging: (&(objectclass=rfc822mailgroup)(cn=*Computer*)) -> num of entries = 84
		TestMultiGoroutineSearch: starting...
		TestMultiGoroutineSearch(0): (cn=cis-fac) -> num of entries = 0
		TestMultiGoroutineSearch(1): (&(owner=*)(cn=cis-fac)) -> num of entries = 0
		TestMultiGoroutineSearch(2): (&(objectclass=rfc822mailgroup)(cn=*Computer*)) -> num of entries = 84
		TestMultiGoroutineSearch(3): (&(objectclass=rfc822mailgroup)(cn=*Mathematics*)) -> num of entries = 18
		TestMultiGoroutineSearch: starting...
		TestMultiGoroutineSearch(0): (cn=cis-fac) -> num of entries = 0
		TestMultiGoroutineSearch(1): (&(owner=*)(cn=cis-fac)) -> num of entries = 0
		TestMultiGoroutineSearch(2): (&(objectclass=rfc822mailgroup)(cn=*Computer*)) -> num of entries = 84
		TestMultiGoroutineSearch(3): (&(objectclass=rfc822mailgroup)(cn=*Mathematics*)) -> num of entries = 18
		TestMultiGoroutineSearch: starting...
		TestMultiGoroutineSearch: using StartTLS...
		TestMultiGoroutineSearch(0): (cn=cis-fac) -> num of entries = 0
		TestMultiGoroutineSearch(1): (&(owner=*)(cn=cis-fac)) -> num of entries = 0
		TestMultiGoroutineSearch(2): (&(objectclass=rfc822mailgroup)(cn=*Computer*)) -> num of entries = 84
		TestMultiGoroutineSearch(3): (&(objectclass=rfc822mailgroup)(cn=*Mathematics*)) -> num of entries = 18
		TestCompare: starting...
		TestCompare: -> num of entries = %!d(bool=true)
		PASS
		ok      src/github.com/BestianRU/ldap   10.615s

### CODE REVIEW

**Comment by BestianRU** [comment]
Checks with cyrillic symbols have passed:
image
image

**Comment by johnweldon** [comment]
@BestianRU very nice, thank you!  Would you be able to add unit tests too?  We can probably get unit tests written after the fact, but if we don't get them in before we merge it may never happen :)

File: filter.go [hunk 1]
```diff
@@ -166,6 +178,12 @@ func compileFilter(filter string, pos int) (*ber.Packet, int, error) {
 		}
 	}()
 
+	xpar_stop , _ := utf8.DecodeRuneInString(")")
```
Comment (liggitt) [inline filter.go]: Can't you just cast `rune(')')`, etc?

Comment (liggitt) [inline filter.go]: If so, keeping the cast literals inline below would be easier to follow, I think

Comment (BestianRU) [inline filter.go]: True remark! I will correct itю

**Comment by BestianRU** [comment]
Add unit-test too - good idea. But i need to study this question :)

**Comment by liggitt** [comment]
@BestianRU I opened https://github.com/BestianRU/ldap/pull/1 as a pull against your branch to add tests and simplify the rune reading... can you check out that branch and verify everything is still working as expected on your end? If you accept the pull request to your branch, it will automatically update this PR

**Comment by liggitt** [comment]
Sorry, one more at https://github.com/BestianRU/ldap/pull/2 ... realized I wasn't returning compile errors correctly, added tests for unclosed filter errors

**Comment by BestianRU** [comment]
Good tests. Good knowledge for me :) Thank you :)

**Comment by liggitt** [comment]
feel free to squash down to a single commit (or two commits, one with your changes and one with mine) before we merge

**Comment by BestianRU** [comment]
You can reopen this pull back? I made a mistake during a merger, but already repaired all.

**Comment by johnweldon** [comment]
@BestianRU I'm afraid you deleted the branch on your repo... you might be able to undelete the branch on your repo?

**Comment by liggitt** [comment]
I have it locally, I can squash and reopen

**Comment by liggitt** [comment]
Opened https://github.com/go-ldap/ldap/pull/34

**Pull request closed without merging**